### DISCUSSION TITLE
Include sambvca binary

### DISCUSSION THREAD

**Discussion by kjelljorner** [JacksonBurns/py2sambvca#22] [Ideas]
I'm not sure if it would be allowed under the license, but you could in principle package a binary version of sambvca with your py2sambvca wheels for pip installation. I made something similar with [libconeangle](https://github.com/kjelljorner/libconeangle/blob/main/.github/workflows/build_wheels.yml), using [scikit-build](https://scikit-build.readthedocs.io/en/latest/), and [simple-dftd3](https://github.com/dftd3/simple-dftd3/blob/main/.github/workflows/wheel.yml) does it with meson. That would take a way a bit of the hassle having to compile it and then point to a certain directory.

**Comment by kjelljorner** [comment]
I checked just for fun, and it seems a bit difficult currently as you need to package a binary and not a library as is most often the case. Tools like delocate and auditwheel don't seem to work with binaries. Conda-forge would still be an option though.

**Reply by JacksonBurns** [reply]
I'll give this a look! I have never tried doing anything like this but it would be of great use. Thanks!

**Reply by kjelljorner** [reply]
Just to clarify, the main problem is that the binary would depend on some runtime libraries, such as `libgfortran` and `libquadmath` if you use GFortran. On some OSs, such as Windows and some Linux distros, you can statically link to these, so that the binary is independent. On MacOS, it would link to them dynamically, and the user would need to have them installed on their system to run the binary (which they probably don't). In the case of libraries, you can "patch" the Python wheels - that is, copy the runtime libraries into the wheels. For binaries you can't apparently. If you distribute on conda-forge, you can make sure that the users also have the runtime libraries by listing Fortran as a dependency. Maybe @awvwgk has some idea of how to package binaries?

**Reply by awvwgk** [reply]
> For binaries you can't apparently

Oh, you certainly can put binaries into wheels. But please, don't do it, that's not the way you are supposed to serve them.

> distribute on conda-forge

That would work for sure. Looking at the SambVca source code, creating a package shouldn't be too hard, but the additional licensing conditions seem like a showstopper for any distribution.
